modules/metastorage/README.md

```diff
@@ -0,0 +1,46 @@
+# Metastorage
+
+The module for store and access to metadata. There is module is connected with this one:
+
+- metastorage-api - the module contains classes to access to metastorage service from another components.
+
+The storage is distributed to prevent losing data and based of RAFT. That distributed storage persists on top of every cluster node, however
+most of such participants are learners and only few of them are voting ones. In typical case, the group consists of only several voted nodes
+of the cluster (amount of nodes have to be odd 3 or 5). The rest of nodes listens metadata update but does not participant in voting (RAFT 
+learners).
+
```

Comment:
To avoid data loss, the storage is distributed using the RAFT consensus algorithm. Every cluster node has access to such distributed storage, but the majority of members are learners, and only a small number are voters in terms of RAFT algorithm. Typically, only a small number of the cluster's voting nodes make up the raft-group (number of nodes must be odd, either 3 or 5.). The remaining nodes listen to metadata updates but do not vote (they called learners in terms of RAFT).